broker/src/test/java/io/zeebe/broker/system/partitions/ZeebePartitionTest.java

```diff
@@ -61,6 +63,41 @@ public void shouldInstallLeaderPartition() {
     verify(transition).toLeader();
   }
 
+  @Test
+  public void shouldCallOnFailureOnAddFailureListenerAndUnhealthy() {
+    // given
+    final ZeebePartition partition = new ZeebePartition(ctx, transition);
+    schedulerRule.submitActor(partition);
+    final FailureListener failureListener = mock(FailureListener.class);
+    doNothing().when(failureListener).onFailure();
+
+    // when
+    partition.addFailureListener(failureListener);
+    schedulerRule.workUntilDone();
+
+    // then
+    verify(failureListener, only()).onFailure();
+  }
+
+  @Test
+  public void shouldCallOnRecoveredOnAddFailureListenerAndHealthy() {
+    // given
+    final ZeebePartition partition = new ZeebePartition(ctx, transition);
+    schedulerRule.submitActor(partition);
+    final FailureListener failureListener = mock(FailureListener.class);
+    doNothing().when(failureListener).onRecovered();
+    // make partition healthy
+    partition.onNewRole(Role.LEADER, 1);
```

Comment:
You can actually just use the `healthMonitor` (see L45), and stub the healthstatus. Something like:
```
when(healthMonitor.getHealthStatus())
  .thenReturn(HealthStatus.HEALTHY)
```

Reply:
Done